src/inspector/models/score/scoreaccessibilitysettingsmodel.cpp

```diff
@@ -0,0 +1,216 @@
+{
+    // Placeholder
+}
+
+void ScoreAccessibilitySettingsModel::loadProperties()
+{
+    // Placeholder
+}
+
+void ScoreAccessibilitySettingsModel::resetProperties()
+{
+    // Placeholder
+}
+
+muse::io::path_t ScoreAccessibilitySettingsModel::getStyleFilePath(ScoreStylePreset preset) const
+{
+    muse::io::path_t basePath = globalConfiguration()->appDataPath() + "styles/MSN/";
+    switch (preset) {
+    case ScoreStylePreset::DEFAULT:
+        return engravingConfiguration()->defaultStyleFilePath();
+    case ScoreStylePreset::MSN_16MM:
+        return basePath + "16mm_MSN.mss";
+    case ScoreStylePreset::MSN_18MM:
+        return basePath + "18mm_MSN.mss";
+    case ScoreStylePreset::MSN_20MM:
+        return basePath + "20mm_MSN.mss";
+    case ScoreStylePreset::MSN_22MM:
+        return basePath + "22mm_MSN.mss";
+    case ScoreStylePreset::MSN_25MM:
+        return basePath + "25mm_MSN.mss";
+    default:
+        return muse::io::path_t();
+    }
+}
+
+QVariantList ScoreAccessibilitySettingsModel::possibleScoreStylePresets() const
+{
+    QVariantList presets;
+
+    QString text = "text";
+    QString value = "value";
+    QString preset = "preset";
+    QString edited = "edited";
+
+    for (int i = 0; i < static_cast<int>(ScoreStylePreset::MAX_PRESET); ++i) {
+        ScoreStylePreset presetEnum = static_cast<ScoreStylePreset>(i);
+        QString presetName = TConv::translatedUserName(presetEnum);
+
+        presets.append(QVariantMap {
+            { text, presetName },
+            {
+                value,
+                QVariantMap {
+                    { preset, i },
+                    { edited, false }
+                }
+            }
+        });
+
+        if (i == static_cast<int>(m_scoreStylePreset) && m_scoreStylePresetEdited) {
+            QString editedPresetName = muse::qtrc("inspector", "%1 (edited)").arg(presetName);
+            presets.append(QVariantMap {
+                { text, editedPresetName },
+                {
+                    value,
+                    QVariantMap {
+                        { preset, i },
+                        { edited, true }
+                    }
+                }
+            });
+        }
+    }
+
+    return presets;
+}
+
+void ScoreAccessibilitySettingsModel::setScoreStylePresetIndex(int index)
+{
+    int oldIndex = scoreStylePresetIndex();
+
+    if (index == oldIndex) {
+        return;
+    }
+
+    if (m_scoreStylePresetEdited && index > oldIndex) {
+        --index;
+    }
+
+    auto selectedPreset = static_cast<ScoreStylePreset>(index);
+
+    bool presetChanged = (selectedPreset != m_scoreStylePreset);
+    bool editedChanged = (m_scoreStylePresetEdited != false);
+
+    MStyle& style = globalContext()->currentNotation()->elements()->msScore()->score()->style();
+
+    if (presetChanged) {
+        m_scoreStylePreset = selectedPreset;
+        loadStyle(selectedPreset);
+        style.setPreset(selectedPreset);
```

Comment:
Please swap these last two lines so it becomes:

```C++
m_scoreStylePreset = selectedPreset;
style.setPreset(selectedPreset);
loadStyle(selectedPreset);
```
Every time you update the cached value, we want to see the real value updated immediately afterwards.

Reply:
This also works:

```C++
loadStyle(selectedPreset);
m_scoreStylePreset = selectedPreset;
style.setPreset(selectedPreset);
```

Or:

```C++
loadStyle(selectedPreset);
style.setPreset(selectedPreset);
m_scoreStylePreset = selectedPreset;
```

What matters is `style.setPreset(selectedPreset);` and `m_scoreStylePreset = selectedPreset;` are on neighbouring lines.

Reply:
I see, yeah makes sense.  I was going forward with the first one. Thanks Peter!